cub/docs/developer_overview.rst

```diff
@@ -714,3 +714,32 @@ we introduced ``cub::detail::temporary_storage::layout``:
     // `allocation_2` alias `allocation_1`, safe to use in stream order
     use(allocation_2.get(), stream); 
 
+
+Symbols visibility
+====================================
+
+Using CUB/Thrust in shared libraries is a known source of issues. 
+For a while, the solution to these issues consisted of wrapping CUB/Thrust namespaces with 
+the ``THRUST_CUB_WRAPPED_NAMESPACE`` macro so that different shared libraries have different symbols. 
+This solution has poor discoverability, 
+since issues present themselves in forms of segmentation faults, hangs, wrong results, etc. 
+To eliminate the symbol visibility issues on our end, we follow the following rules: 
+
+    #. Hiding kernel launchers: 
+       it's important that kernel launchers like Thrust ``triple_chevron`` always reside in the same 
+       library as the API using these kernel launchers.  
+
+    #. Hiding all kernels: 
+       it's important that kernels always reside in the same library as the API using these kernels.  
+
+    #. Incorporating GPU architectures into symbol names: 
+       it's important that kernels compiled for a given GPU architecture are always used by the host 
+       API compiled for that architecture.  
+
+To satisfy (1), ``thrust::cuda_cub::launcher::triple_chevron`` visibility is hidden.  
+
+To satisfy (2), instead of annotating kernels as ``__global__`` we annotate them as 
+``CUB_DETAIL_KERNEL_ATTRIBUTES``. 
```

Comment:
Perhaps add: `CUB_DETAIL_KERNEL_ATTRIBUTES` means `__global__` and `__attribute__ ((visibility ("hidden")))`. 

Right now, it is not entirely clear how `CUB_DETAIL_KERNEL_ATTRIBUTES` satisfies point(2).